gluten-core/src/main/scala/io/glutenproject/extension/ColumnarQueryStagePrepOverrides.scala

```diff
@@ -89,7 +89,6 @@ case class FallbackBroadcastExchange(session: SparkSession) extends Rule[SparkPl
 
 object ColumnarQueryStagePrepOverrides extends GlutenSparkExtensionsInjector {
   override def inject(extensions: SparkSessionExtensions): Unit = {
-    val builders = TagBeforeTransformHits.ruleBuilders :+ FallbackBroadcastExchange
-    builders.foreach(extensions.injectQueryStagePrepRule)
+    extensions.injectQueryStagePrepRule(FallbackBroadcastExchange)
```

Comment:
I think the cost of stagePrepRule is bigger than columnarRule because AQE re-optimize would produce some ineffective results and fallback to original plan. So we can avoid applying unnecessary rules after moving to columnarRule.